engine/v2/transaction/beef/beef_service.go

```diff
@@ -0,0 +1,95 @@
+package beef
+
+import (
+	"context"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/v2/transaction/errors"
+)
+
+// TxQueryResult represents the result of a transaction query.
+type TxQueryResult struct {
+	SourceTXID string  // Source Transaction ID.
+	RawHex     *string // Raw transaction in hexadecimal format.
+	BeefHex    *string // BEEF-formatted transaction.
+}
+
+// IsBeef checks if the TxQueryResult contains a BEEF-formatted transaction.
+func (tx TxQueryResult) IsBeef() bool { return tx.BeefHex != nil }
+
+// IsRawTx checks if the TxQueryResult contains a raw transaction.
+func (tx TxQueryResult) IsRawTx() bool { return tx.RawHex != nil }
+
+// TxRepository defines an interface for querying transaction input sources.
+// It provides a method to retrieve transaction details for a given set of source transaction IDs.
+type TxRepository interface {
+	// QueryInputSources retrieves transaction query results for the provided source transaction IDs.
+	// Returns a slice of TxQueryResult and an error if the query fails.
+	QueryInputSources(ctx context.Context, sourceTXIDs ...string) (TxQueryResultSlice, error)
+}
+
+// Service provides transaction processing functionalities, including preparing BEEF-encoded transactions.
+type Service struct {
+	repository TxRepository // Repository used to query transaction input sources.
+}
+
+// extractSourceTXIDs extracts source transaction IDs from the inputs of the given transaction.
+// It returns a slice of transaction IDs or an error if the transaction has no inputs.
+func (s *Service) extractSourceTXIDs(tx *sdk.Transaction) ([]string, error) {
+	if tx.InputCount() == 0 {
+		return nil, txerrors.ErrZeroInputCount
+	}
+
+	sourceTXIDs := make([]string, len(tx.Inputs))
```

Comment:
I am not sure but doesn't mixing prealloacting with "append" function cause memory leaks sometimes?

What I mean:
```go
foos := make([]string, len(items))
for _, item := range items {
      foos = append(foos, "Foo" + item) // Appending to pre-allocated slice
}
```

to be replaced with:
```go
foos := make([]string, len(items))
for i, item := range items {
     foos[i] = "Foo" + item // Direct assignment using index
}
```